lib/mixlib/install/script_generator.rb

```diff
@@ -0,0 +1,217 @@
+#
+# Author:: Thom May (<thom@chef.io>)
+# Author:: Patrick Wright (<patrick@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "mixlib/install/util"
+require "cgi"
+
+module Mixlib
+  class Install
+    class ScriptGenerator
+      attr_accessor :version
+
+      attr_accessor :powershell
+
+      attr_accessor :prerelease
+
+      attr_accessor :nightlies
```

Comment:
This class used be in Mixlib::Install.  We just moved it as is to the
existing logic would still work.  This will be refactored to generate the
install scripts, and the old omnitruck configurations will go away.

On Fri, Oct 9, 2015 at 8:06 AM, Seth Chisamore <notifications@github.com>
wrote:

> In lib/mixlib/install/script_generator.rb
> <https://github.com/chef/mixlib-install/pull/11#discussion_r41640250>:
>
> > +# limitations under the License.
> > +#
> > +
> > +require "mixlib/install/util"
> > +require "cgi"
> > +
> > +module Mixlib
> > +  class Install
> > +    class ScriptGenerator
> > +      attr_accessor :version
> > +
> > +      attr_accessor :powershell
> > +
> > +      attr_accessor :prerelease
> > +
> > +      attr_accessor :nightlies
>
> IMO we should not support the :prerelease and :nightlies options.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/mixlib-install/pull/11/files#r41640250>.
>